autowarefoundation/autoware.universe
# feat(default_ad_api): add routing api

Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Introduce AD API for routing. This PR is the Adapter and AD API part of the diagram. 
Here are the changes for each package:
- autoware_ad_api_msgs: Add the routing API messages.
- autoware_ad_api_specs: Add the routing API definitions.
- component_interface_specs; Add the internal routing interface definitions.
- component_interface_utils: Some fix for this PR.
- tier4_autoware_api_launch: Add launching the API node.
- default_ad_api: Add the routing API.
- default_ad_api_helpers: Add the adaptor for RViz routing topics.

![planning-routing drawio](https://user-images.githubusercontent.com/43976882/182304917-550251d5-eefb-4182-8f20-b239eb48057b.png)


## Related links

https://github.com/autowarefoundation/autoware.universe/pull/1495
https://github.com/autowarefoundation/autoware_launch/pull/70

## Tests performed

1. Merge these PRs.
    - https://github.com/autowarefoundation/autoware.universe/pull/1495
    - https://github.com/autowarefoundation/autoware_launch/pull/70
2. Launch planning simulation.
3. Set the goal pose and checkpoints from rviz.
4. Check if the vehicle drives well.

## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
